includes/API_Credentials/API_Credentials_Manager.php

```diff
@@ -0,0 +1,270 @@
+ *
+ * @since n.e.x.t
+ * @package wp-ai-client
+ */
+
+namespace WordPress\AI_Client\API_Credentials;
+
+use WordPress\AiClient\AiClient;
+use WordPress\AiClient\Providers\DTO\ProviderMetadata;
+use WordPress\AiClient\Providers\Http\DTO\ApiKeyRequestAuthentication;
+
+/**
+ * Class for managing the AI API credentials for the various providers.
+ *
+ * @since n.e.x.t
+ *
+ * @phpstan-import-type ProviderMetadataArrayShape from ProviderMetadata
+ */
+class API_Credentials_Manager {
+
+	private const OPTION_GROUP                = 'wp-ai-client-settings';
+	private const OPTION_PROVIDER_CREDENTIALS = 'wp_ai_client_provider_credentials';
+
+	/**
+	 * Initializes the API credentials manager.
+	 *
+	 * This method needs to be called by the consumer of this package, on the WordPress 'init' action hook.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function initialize(): void {
+		$this->collect_providers();
+		$this->register_settings();
+		$this->pass_credentials_to_client();
+
+		add_action(
+			'admin_menu',
+			function () {
+				$this->add_admin_screen();
+			}
+		);
+	}
+
+	/**
+	 * Collects metadata for all registered providers in the PHP AI Client SDK.
+	 *
+	 * Since the PHP AI Client SDK as well as the WordPress AI Client package can be loaded multiple times,
+	 * including with different namespace or class name prefixes, this method ensures that the provider metadata is
+	 * collected only once across all instances of the package.
+	 *
+	 * This unified collection mechanism allows the WordPress AI Client package to expose a single settings screen in
+	 * WordPress for managing API credentials for all providers, regardless of how many times the package is loaded and
+	 * regardless of whether a provider is only registered in one of the instances.
+	 *
+	 * To safely do that, the method uses a global variable. It stores the provider metadata array keyed by the
+	 * provider ID, and for each provider metadata it also stores a map of the AiClient class names where the provider
+	 * is registered in.
+	 *
+	 * @since n.e.x.t
+	 */
+	private function collect_providers(): void {
+		/**
+		 * The internal global, to collect providers metadata across duplicate clients, including prefixed versions.
+		 *
+		 * @var ?array<string, ProviderMetadataArrayShape> $wp_ai_client_providers_metadata
+		 */
+		global $wp_ai_client_providers_metadata;
+
+		if ( ! isset( $wp_ai_client_providers_metadata ) ) {
+			$wp_ai_client_providers_metadata = array();
+		}
+
+		$registry = AiClient::defaultRegistry();
+
+		$provider_ids = $registry->getRegisteredProviderIds();
+		foreach ( $provider_ids as $provider_id ) {
+			if ( ! $registry->hasProvider( $provider_id ) ) {
+				continue;
+			}
+
+			// If the provider was already found via another client class, just this client class name to the list.
+			if ( isset( $wp_ai_client_providers_metadata[ $provider_id ] ) ) {
+				$wp_ai_client_providers_metadata[ $provider_id ]['ai_client_classnames'][ AiClient::class ] = true;
```

Comment:
I don't think this fits the shape of the array defined above for this variable.

Reply:
Fixed in https://github.com/WordPress/wp-ai-client/pull/6/commits/4dacba053745104de1d1b2e299edb8d973fc103f, but for some reason that I can't figure out PHPStan is complaining. Do you have any hunch?

Reply:
It seems like it's having trouble resolving the imported type. I'd double check the imported type name, and maybe try using a FQCN in the import.

Reply:
Just tested, the import itself is working fine. I have no idea what's going on here. If I can't resolve this, I'm just going to redeclare the whole thing for now.